sdk/azidentity/azidentity_test.go

```diff
@@ -23,10 +28,114 @@ const (
-	customHostString = "https://custommock.com/"
+	envHostString              = "https://mock.com/"
+	customHostString           = "https://custommock.com/"
+	getTenantDiscoveryResponse = `{
+		"token_endpoint": "https://login.microsoftonline.com/3c631bb7-a9f7-4343-a5ba-a6159135f1fc/oauth2/v2.0/token",
+		"token_endpoint_auth_methods_supported": [
+		"client_secret_post",
+		"private_key_jwt",
+		"client_secret_basic"
+		],
+		"jwks_uri": "https://login.microsoftonline.com/3c631bb7-a9f7-4343-a5ba-a6159135f1fc/discovery/v2.0/keys",
+		"response_modes_supported": [
+		"query",
+		"fragment",
+		"form_post"
+		],
+		"subject_types_supported": [
+		"pairwise"
+		],
+		"id_token_signing_alg_values_supported": [
+		"RS256"
+		],
+		"response_types_supported": [
+		"code",
+		"id_token",
+		"code id_token",
+		"id_token token"
+		],
+		"scopes_supported": [
+		"openid",
+		"profile",
+		"email",
+		"offline_access"
+		],
+		"issuer": "https://login.microsoftonline.com/3c631bb7-a9f7-4343-a5ba-a6159135f1fc/v2.0",
+		"request_uri_parameter_supported": false,
+		"userinfo_endpoint": "https://graph.microsoft.com/oidc/userinfo",
+		"authorization_endpoint": "https://login.microsoftonline.com/3c631bb7-a9f7-4343-a5ba-a6159135f1fc/oauth2/v2.0/authorize",
+		"device_authorization_endpoint": "https://login.microsoftonline.com/3c631bb7-a9f7-4343-a5ba-a6159135f1fc/oauth2/v2.0/devicecode",
+		"http_logout_supported": true,
+		"frontchannel_logout_supported": true,
+		"end_session_endpoint": "https://login.microsoftonline.com/3c631bb7-a9f7-4343-a5ba-a6159135f1fc/oauth2/v2.0/logout",
+		"claims_supported": [
+		"sub",
+		"iss",
+		"cloud_instance_name",
+		"cloud_instance_host_name",
+		"cloud_graph_host_name",
+		"msgraph_host",
+		"aud",
+		"exp",
+		"iat",
+		"auth_time",
+		"acr",
+		"nonce",
+		"preferred_username",
+		"name",
+		"tid",
+		"ver",
+		"at_hash",
+		"c_hash",
+		"email"
+		],
+		"kerberos_endpoint": "https://login.microsoftonline.com/3c631bb7-a9f7-4343-a5ba-a6159135f1fc/kerberos",
+		"tenant_region_scope": "NA",
+		"cloud_instance_name": "microsoftonline.com",
+		"cloud_graph_host_name": "graph.windows.net",
+		"msgraph_host": "graph.microsoft.com",
+		"rbac_url": "https://pas.windows.net"
+		}
+		},
+		{
+		"RequestUri": "https://login.microsoftonline.com/fake-tenant/oauth2/v2.0/token",
+		"RequestMethod": "POST",
+		"RequestHeaders": {
+		":authority": "localhost:5001",
+		":method": "POST",
+		":path": "/fake-tenant/oauth2/v2.0/token",
+		":scheme": "https",
+		"Accept-Encoding": "gzip",
+		"Content-Length": "2",
+		"Content-Type": "application/x-www-form-urlencoded; charset=utf-8",
+		"User-Agent": "azsdk-go-azidentity/v0.12.1 azsdk-go-azcore/v0.21.0 (go1.16.7; linux)"
+		},
+		"RequestBody": {},
+		"StatusCode": 400,`
 )
 
+func validateJWTRequestContainsHeader(t *testing.T, headerName string) mock.ResponsePredicate {
+	return func(req *http.Request) bool {
+		body, err := ioutil.ReadAll(req.Body)
+		if err == nil {
```

Comment:
Nitpicking: `err != nil` would be more idiomatic (handler immediately follows the call that returned the error; happy path less indented)